addons/pos_self_order/static/src/kiosk/pages/product_list/product_list.js

```diff
@@ -0,0 +1,92 @@
+import { useselfOrder } from "@pos_self_order/kiosk/self_order_kiosk_service";
+import { ProductCard } from "@pos_self_order/kiosk/components/product_card/product_card";
+import { CancelPopup } from "@pos_self_order/kiosk/components/cancel_popup/cancel_popup";
+import { KioskTemplate } from "@pos_self_order/kiosk/template/kiosk_template";
+import { useDetection } from "@pos_self_order/common/hooks/use_detection";
+import { useService, useChildRef } from "@web/core/utils/hooks";
+import { _t } from "@web/core/l10n/translation";
+
+export class ProductList extends Component {
+    static template = "pos_self_order.ProductList";
+    static components = { ProductCard, KioskTemplate };
+
+    setup() {
+        this.selfOrder = useselfOrder();
+        this.dialog = useService("dialog");
+        this.router = useService("router");
+        this.productsList = useRef("productsList");
+        this.categoryList = useRef("categoryList");
+        this.currentProductCard = useChildRef();
+
+        this.productGroups = Object.fromEntries(
+            Array.from(this.selfOrder.categoryList).map((category) => {
+                return [category.name, useRef(`productsWithCategory_${category.name}`)];
+            })
+        );
+
+        this.categoryButton = Object.fromEntries(
+            Array.from(this.selfOrder.categoryList).map((category) => {
+                return [category.name, useRef(`category_${category.name}`)];
+            })
+        );
+
+        this.currentProductGroup = useDetection(this.productsList, this.productGroups, () => []);
+
+        useEffect(
+            () => {
+                const category = this.categoryButton[this.currentProductGroup.name]?.el;
+
+                if (!category) {
+                    return;
+                }
+
+                this.categoryList.el.scroll({
+                    left: category.offsetLeft + category.offsetWidth / 2 - window.innerWidth / 2,
+                    behavior: "smooth",
+                });
+            },
+            () => [this.currentProductGroup.name]
+        );
+
+        onMounted(() => {
+            if (this.selfOrder.lastEditedProductId) {
+                this.scrollTo(this.currentProductCard, { behavior: "instant" });
+            }
+        });
+    }
+
+    categoryClicked(category) {
+        if (category == "Combos") {
+            console.log("combos");
+        }
+        this.scrollTo(
+            this.currentProductGroup.name != category ? this.productGroups[category] : null
+        );
+    }
+
+    scrollTo(ref = null, { behavior = "smooth" } = {}) {
+        this.productsList.el.scroll({
+            top: ref?.el ? ref.el.offsetTop - this.productsList.el.offsetTop : 0,
+            behavior,
+        });
+    }
+
+    review() {
+        this.router.navigate("cart");
+    }
+
+    cancelOrder() {
+        this.dialog.add(CancelPopup, {
+            title: _t("Cancel order"),
+            body: _t(
```

Comment:
The CancelPopup does not use body as props. Either remove the body props or change the CancelPopup to show a props.body